File: pc_shop_frontend/src/models/Shipment.ts
import Reseller from './Reseller'
import Shipment_item from './Shipment_item'

export default class Shipment {
  id: Number
  date: Date
  reseller: Reseller
  shipment_items: any[]

  constructor(data: any) {
    debugger
    this.id = data.id
    this.date = data.date
    this.reseller = data.reseller
    this.shipment_items = []
    for (let shipment_item of data.shipment_items)
      this.shipment_items.push(new Shipment_item(shipment_item))
  }
}

File: pc_shop_backend/src/main/java/com/example/pc_shop_backend/model/Shipment.java
package com.example.pc_shop_backend.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.Date;
import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class Shipment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    private Date date;

    @ManyToOne
    @JoinColumn(name = "reseller_id")
    private Reseller reseller;

    @OneToMany(mappedBy = "shipment", cascade = CascadeType.ALL)
    private List<Shipment_item> shipment_items;

}
